### ISSUE TITLE
TextWriter write text in a wrong position

### ISSUE THREAD

**Issue by hgdhot** [pymupdf/PyMuPDF#2246]
## Describe the bug (mandatory)
TextWriter write text in a wrong position

## To Reproduce (mandatory)
```
import fitz

document = fitz.open("./11100002.pdf")
page = document.load_page(1)

writer = fitz.TextWriter(page.rect)
redact_area = fitz.Rect(31.921897888183594, 53.70486068725586, 283.53240966796875, 117.14781951904297)
page.add_redact_annot(redact_area)
page.apply_redactions(images=1)

pos = fitz.Point(34.1218978881836, 65.52986121177673)
writer.append(pos, "The brown fox jump over the...")

shape = page.new_shape()
shape.draw_rect(writer.text_rect)  
shape.finish(width=0.3, color=(1,0,0), fill=(1,0,0))
shape.commit()

writer.write_text(page)
document.save('test.pdf', garbage=3)
```

## Expected behavior (optional)
The "The brown fox jump over the..." show within writer.text_rect

## Screenshots (optional)
![9be728cbc27d99a883e983ba9dde819b](https://user-images.githubusercontent.com/19300626/220579179-7ba0d72c-a3d6-44ab-b779-6c8b30661bea.png)

## Your configuration (mandatory)
 - Centos 6.3 64-bit
 - Python 3.7.6
 - PyMuPDF==1.21.1

## Additional context (optional)
[11100002.pdf](https://github.com/pymupdf/PyMuPDF/files/10802075/11100002.pdf)

**Comment by JorjMcKie** [comment]
I am underway currently, so just briefly a recommendation without having looked at details: execute page.clean_contents() immediately after loading the page.
Fat chance that this the reason. Please also have a look at the documentation under "misplaced insertions". This probably is no bug.

**Comment by hgdhot** [comment]
Thanks for the quick response!
I have read the documentation under "misplaced insertions", and tried to execute page.clean_contents()，but it doesn't work，the written text still appears in the wrong place。
I found that the crapbox and medibox on this page are not equal. Could it be related to this problem？
![0191028906da5dc2523cbb2c5c993b22](https://user-images.githubusercontent.com/19300626/220836050-7e3880da-6c26-4341-84cf-335fa32efa29.png)

**Comment by julian-smith-artifex-com** [comment]
@JorjMcKie is away at the moment, so i've taken a look at this.

I think your script can be made to work by adding/subtracting `page.mediabox.y0` to some of the `y` coordinates:

```
import fitz

document = fitz.open("./11100002.pdf")
page = document.load_page(1)

writer = fitz.TextWriter(page.rect)
redact_area = fitz.Rect(31.921897888183594, 53.70486068725586, 283.53240966796875, 117.14781951904297)
page.add_redact_annot(redact_area)
page.apply_redactions(images=1)

pos = fitz.Point(34.1218978881836, 65.52986121177673)
writer.append(fitz.Point(pos.x, pos.y-page.mediabox.y0), "The brown fox jump over the...")

shape = page.new_shape()
rect = writer.text_rect
rect.y0 += page.mediabox.y0
rect.y1 += page.mediabox.y0
shape.draw_rect(rect)
shape.finish(width=0.3, color=(1,0,0), fill=(1,0,0))
shape.commit()

writer.write_text(page)
document.save('test.pdf', garbage=3)
```

I'll try to come up with a useful explanation for why this works, but i suspect @JorjMcKie will be able to explain things much better than me.

**Comment by hgdhot** [comment]
@julian-smith-artifex-com Thanks for your replay!
after adding/subtracting page.mediabox.y0, the script worked! So is this misplaced insertions related to the media box(not equal to cropbox)?
I wonder whether I can insert text in this way, regardless of whether the crapbox is equal to the media box?

**Comment by julian-smith-artifex-com** [comment]
Yes it's related to the difference between the coordinate systems for mediabox and cropbox. I hesitate to say more because i don't know enough about PDF/PyMuPDF coordinate systems.

**Comment by JorjMcKie** [comment]
There is an error computed the correct y-ccordinate in TextWriter.
Apparently only there, because `page.draw_rect()` and `page.insert_text()` both work correctly. Looking into this now.

**Comment by julian-smith-artifex-com** [comment]
This is fixed in 1.22.0.
